### PR TITLE
test(mempool): add tests for commit block method

### PR DESCRIPTION
**Pull request by MohammadNassar1**
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/mempool/396)

### CODE REVIEW

**Review by MohammadNassar1 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/mempool/396)* status: 0 of 1 files reviewed, 1 unresolved discussion

*[a discussion](https://reviewable.io/reviews/starkware-libs/mempool/396#-O1Gp17n6_ThNoGEJHBf:-O1Gp17n6_ThNoGEJHBg:b-mtfgcx) (no related file):*
This PR adds tests for the `commit_block` method across various scenarios. 
Since the method is not yet implemented, the tests are currently marked with `#[ignore]` to prevent them from running.

___

**Review by elintul — COMMENTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/mempool/396)* status: 0 of 1 files reviewed, 1 unresolved discussion

**Review by MohammadNassar1 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/mempool/396)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @elintul)

*[a discussion](https://reviewable.io/reviews/starkware-libs/mempool/396#-O1Gp17n6_ThNoGEJHBf:-O2P3Iej4Uruhbmjnbum:bfu0b7v) (no related file):*
<details><summary><i>Previously, MohammadNassar1 (mohammad-starkware) wrote…</i></summary><blockquote>

This PR adds tests for the `commit_block` method across various scenarios. 
Since the method is not yet implemented, the tests are currently marked with `#[ignore]` to prevent them from running.
</blockquote></details>

Now, it adds a single test: a test without rejected or reverted transactions.

___

**Review by elintul — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/mempool/396)* status: all files reviewed, 7 unresolved discussions (waiting on @MohammadNassar1)

___
*[`crates/mempool/src/mempool_test.rs` line 427 at r3](https://reviewable.io/reviews/starkware-libs/mempool/396#-O2TxOZz-W2fB-dAT58U:-O2TxOZz-W2fB-dAT58V:byupo3x) ([raw file](https://github.com/starkware-libs/mempool/blob/1b07b23c8812554540c8a8296f7b00f94f26e7a9/crates/mempool/src/mempool_test.rs#L427)):*
> ```XML
> #[rstest]
> #[ignore]
> fn test_commit_block_includes_all_transactions() {
> ```

Not sure the test case is clear for me from the name. Do you mean you test the case where the Q shouldn't be changed during commit block?



_Code quote:_
```XML
test_commit_block_includes_all_transactions
```
___
*[`crates/mempool/src/mempool_test.rs` line 428 at r3](https://reviewable.io/reviews/starkware-libs/mempool/396#-O2TwcFbDNLP4u4BMiJQ:-O2TwcFbDNLP4u4BMiJR:undefined) ([raw file](https://github.com/starkware-libs/mempool/blob/1b07b23c8812554540c8a8296f7b00f94f26e7a9/crates/mempool/src/mempool_test.rs#L428)):*
> ```XML
> #[ignore]
> fn test_commit_block_includes_all_transactions() {
>     let tx_address0_nonce4 = add_tx_input!(tip: 4, tx_hash: 1, sender_address: "0x0", tx_nonce: 4_u8, account_nonce: 4_u8).tx;
> ```





_Suggestion:_
```XML
fn test_commit_block_includes_all_transactions() {
    // Setup.
    let tx_address0_nonce4 = add_tx_input!(tip: 4, tx_hash: 1, sender_address: "0x0", tx_nonce: 4_u8, account_nonce: 4_u8).tx;
```
___
*[`crates/mempool/src/mempool_test.rs` line 431 at r3](https://reviewable.io/reviews/starkware-libs/mempool/396#-O2Twa322IswTFVD7TfW:-O2Twa322IswTFVD7TfX:undefined) ([raw file](https://github.com/starkware-libs/mempool/blob/1b07b23c8812554540c8a8296f7b00f94f26e7a9/crates/mempool/src/mempool_test.rs#L431)):*
> ```XML
>     let tx_address0_nonce5 = add_tx_input!(tip: 3, tx_hash: 2, sender_address: "0x0", tx_nonce: 5_u8, account_nonce: 4_u8).tx;
>     let tx_address1_nonce3 = add_tx_input!(tip: 2, tx_hash: 3, sender_address: "0x1", tx_nonce: 3_u8, account_nonce: 3_u8).tx;
>     let tx_address2_nonce3 = add_tx_input!(tip: 1, tx_hash: 4, sender_address: "0x2", tx_nonce: 1_u8, account_nonce: 1_u8).tx;
> ```





_Suggestion:_
```XML
tx_address2_nonce1
```
___
*[`crates/mempool/src/mempool_test.rs` line 445 at r3](https://reviewable.io/reviews/starkware-libs/mempool/396#-O2TwyIs9uZggbuqhYQy:-O2TwyIs9uZggbuqhYQz:bwfy49) ([raw file](https://github.com/starkware-libs/mempool/blob/1b07b23c8812554540c8a8296f7b00f94f26e7a9/crates/mempool/src/mempool_test.rs#L445)):*
> ```XML
>     let txs_iterator = pool_txs.iter().cloned();
> 
>     let mut mempool: Mempool = MempoolState::new(txs_iterator, tx_references_iterator).into();
> ```

Why is `.cloned()` needed for `txs_iterator`? Can it be avoided?



_Suggestion:_
```XML
    ];
    let tx_references_iterator = queued_txs.iter().map(TransactionReference::new);
    let txs_iterator = pool_txs.iter().cloned();
    let mut mempool: Mempool = MempoolState::new(txs_iterator, tx_references_iterator).into();
```
___
*[`crates/mempool/src/mempool_test.rs` line 447 at r3](https://reviewable.io/reviews/starkware-libs/mempool/396#-O2Tx6m92wDx0E22rQn2:-O2Tx6m92wDx0E22rQn3:undefined) ([raw file](https://github.com/starkware-libs/mempool/blob/1b07b23c8812554540c8a8296f7b00f94f26e7a9/crates/mempool/src/mempool_test.rs#L447)):*
> ```XML
>     let mut mempool: Mempool = MempoolState::new(txs_iterator, tx_references_iterator).into();
> 
>     let commit_state = HashMap::from([
> ```





_Suggestion:_
```XML
// Test.
let state_changes = HashMap::from([
```
___
*[`crates/mempool/src/mempool_test.rs` line 453 at r3](https://reviewable.io/reviews/starkware-libs/mempool/396#-O2TxA23B_BQAXt6yftS:-O2TxA23B_BQAXt6yftT:undefined) ([raw file](https://github.com/starkware-libs/mempool/blob/1b07b23c8812554540c8a8296f7b00f94f26e7a9/crates/mempool/src/mempool_test.rs#L453)):*
> ```XML
>     assert!(mempool.commit_block(commit_state).is_ok());
> 
>     assert_eq_mempool_queue(&mempool, &queued_txs)
> ```





_Suggestion:_
```XML
// Assert.
assert_eq_mempool_queue(&mempool, &queued_txs)
```

**Review by elintul — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/mempool/396)* status: all files reviewed, 7 unresolved discussions (waiting on @MohammadNassar1)

___
*[`crates/mempool/src/mempool_test.rs` line 445 at r3](https://reviewable.io/reviews/starkware-libs/mempool/396#-O2TwyIs9uZggbuqhYQy:-O2UBkTqCaJCLyTE-RXz:b-el92rt) ([raw file](https://github.com/starkware-libs/mempool/blob/1b07b23c8812554540c8a8296f7b00f94f26e7a9/crates/mempool/src/mempool_test.rs#L445)):*
<details><summary><i>Previously, elintul (Elin) wrote…</i></summary><blockquote>

Why is `.cloned()` needed for `txs_iterator`? Can it be avoided?
</blockquote></details>

In other words: how can we reduce transaction cloning to once per test?

**Review by MohammadNassar1 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/mempool/396)* status: 0 of 1 files reviewed, 4 unresolved discussions (waiting on @elintul)

___
*[`crates/mempool/src/mempool_test.rs` line 427 at r3](https://reviewable.io/reviews/starkware-libs/mempool/396#-O2TxOZz-W2fB-dAT58U:-O2Y9VIW-zR4zr5jwda8:bql78r4) ([raw file](https://github.com/starkware-libs/mempool/blob/1b07b23c8812554540c8a8296f7b00f94f26e7a9/crates/mempool/src/mempool_test.rs#L427)):*
<details><summary><i>Previously, elintul (Elin) wrote…</i></summary><blockquote>

Not sure the test case is clear for me from the name. Do you mean you test the case where the Q shouldn't be changed during commit block?
</blockquote></details>

Exactly!
If the Batcher includes all the sent transactions, the block state would be equal to the mempool state, and thus, the Q wouldn't be changed. 
___
*[`crates/mempool/src/mempool_test.rs` line 453 at r3](https://reviewable.io/reviews/starkware-libs/mempool/396#-O2TxA23B_BQAXt6yftS:-O2YD-YYDSkU-aUmB91R:b-896fix) ([raw file](https://github.com/starkware-libs/mempool/blob/1b07b23c8812554540c8a8296f7b00f94f26e7a9/crates/mempool/src/mempool_test.rs#L453)):*
> ```XML
>     assert!(mempool.commit_block(commit_state).is_ok());
> 
>     assert_eq_mempool_queue(&mempool, &queued_txs)
> ```

Done.

**Review by elintul — COMMENTED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/mempool/396)* status: all files reviewed, 3 unresolved discussions (waiting on @MohammadNassar1)

___
*[`crates/mempool/src/mempool_test.rs` line 447 at r3](https://reviewable.io/reviews/starkware-libs/mempool/396#-O2Tx6m92wDx0E22rQn2:-O2YTFqF88uv3iVhNpbt:b-dv65zk) ([raw file](https://github.com/starkware-libs/mempool/blob/1b07b23c8812554540c8a8296f7b00f94f26e7a9/crates/mempool/src/mempool_test.rs#L447)):*
> ```XML
>     let mut mempool: Mempool = MempoolState::new(txs_iterator, tx_references_iterator).into();
> 
>     let commit_state = HashMap::from([
> ```

Please add the `Test` and `Assert` comments?
For structural consistency between tests.

**Review by MohammadNassar1 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/mempool/396)* status: 0 of 1 files reviewed, 3 unresolved discussions (waiting on @elintul)

___
*[`crates/mempool/src/mempool_test.rs` line 445 at r3](https://reviewable.io/reviews/starkware-libs/mempool/396#-O2TwyIs9uZggbuqhYQy:-O2YfaQH2K2burDrWvPk:b-896fix) ([raw file](https://github.com/starkware-libs/mempool/blob/1b07b23c8812554540c8a8296f7b00f94f26e7a9/crates/mempool/src/mempool_test.rs#L445)):*
<details><summary><i>Previously, elintul (Elin) wrote…</i></summary><blockquote>

In other words: how can we reduce transaction cloning to once per test?
</blockquote></details>

Done.

**Review by elintul — COMMENTED** [review]
> Reviewed 1 of 1 files at r5.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/mempool/396)* status: all files reviewed, 3 unresolved discussions (waiting on @MohammadNassar1)

**Review by elintul — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/mempool/396)* status: all files reviewed, 3 unresolved discussions (waiting on @MohammadNassar1)

___
*[`crates/mempool/src/mempool_test.rs` line 427 at r5](https://reviewable.io/reviews/starkware-libs/mempool/396#-O2Za5Mw2H0reTYhq37y:-O2Za5Mw2H0reTYhq37z:undefined) ([raw file](https://github.com/starkware-libs/mempool/blob/4ae3ba6be059fad65af2450b9797bb5931c1710b/crates/mempool/src/mempool_test.rs#L427)):*
> ```XML
> #[rstest]
> #[ignore]
> fn test_commit_block_includes_all_transactions() {
> ```





_Suggestion:_
```XML
all_txs
```

**Review by elintul — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/mempool/396#-:-O2ZqAfg8lBvp7bv2YUR:bnfp4nl)* status: all files reviewed, 1 unresolved discussion

**Review by elintul — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/mempool/396)* status: all files reviewed, 1 unresolved discussion (waiting on @MohammadNassar1)

*[a discussion](https://reviewable.io/reviews/starkware-libs/mempool/396#-O2czEUA7GmPTwBEIYcE:-O2czEUA7GmPTwBEIYcF:bj6opre) (no related file):*
Please update commit message.

___
